### PR TITLE
Feature association

### PR DESCRIPTION
**Pull request by GOlivierNation**
**### What is done in this PR:**
************************************
    Create a class Classroom with the following:
        @label instance variable, should be initialized in the constructor.
        Setter and getter for @label (remember about attr_accessor).
    Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
        Create the has-many side (a classroom has many students).
        Create the belongs-to side (a student belongs to a classroom).
        Make sure that when adding a student to a classroom it also sets the classroom for the student.
        Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
    Create a class Book with the following:
        @title and @author instance variables, should be initialized in the constructor.
        Setters and getters for instance variables (remember about attr_accessor).
    Create a class Rental with the following:
        @date instance variable, should be initialized in the constructor.
        Setter and getter for @date (remember about attr_accessor).
    Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
        Create the has-many side of Book and Rental (a book has many rentals).
        Create the belongs-to side of Rental and Book (a rental belongs to a book).
        Create the has-many side of Person and Rental (a person has many rentals).
        Create the belongs-to side of Rental and Person (a rental belongs to a person).
        Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+class Rental
+  attr_accessor :date
+
+  def initialize(book, person, date = DateTime.now())
+    @date = date
+    @book = book
+    @person = person
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline rental.rb]: - [ ] Kindly add person and book to your attr_accesor

File: student.rb [hunk 2]
```diff
@@ -1,15 +1,17 @@
-require './person'
+require_relative('./person')
 
 class Student < Person
-  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission: parent_permission)
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(name, age, parent_permission)
     @classroom = classroom
```
Comment (VICTOR-CHUKWU) [inline student.rb:8]: - [ ] Kindly make classroom accessible by using attr_accesor

File: person.rb [hunk 3]
```diff
@@ -1,27 +1,33 @@
...
 
-  def correct_name
-    @name
+  def validate_name
+    @name = validate.correct_name(@name)
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (VICTOR-CHUKWU) [inline person.rb]: - [ ] this method should take two parameters book and date , kindly fix this
```
 def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @GOlivierNation 👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @GOlivierNation, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**